src/helpers/ipfs.ts

```diff
@@ -1,4 +1,20 @@
+import { IPFS_TIMEOUT } from '../constants'
 import { create } from 'ipfs-http-client'
 import env from '../helpers/env'
 
-export default create({ url: env.IPFS })
+const ipfs = create({ url: env.IPFS })
+
+export async function getDataFromIPFS(cid: string): Promise<ArrayBuffer> {
+  try {
+    const chunks = []
+    for await (const chunk of ipfs.cat(cid, { timeout: IPFS_TIMEOUT }))
+      chunks.push(...chunk)
+    if (chunks.length === 0) throw new Error("Couldn't get data from IPFS")
+    const data = new Uint8Array(chunks.flat())
+    return data.buffer
+  } catch (e) {
+    throw new Error("Couldn't get data from IPFS")
```

Comment:
Do we need a `catch` then?

Reply:
As I can see we don't you handle everything in the controller, simply throw on `L12` and you're good

Reply:
I wanted to set the custom message so we knew to handle it with 404.